pkg/datapath/iptables/iptables.go

```
@@ -694,6 +695,28 @@ func (m *IptablesManager) InstallProxyRules(proxyPort uint16, ingress bool, name
 	return m.iptProxyRules("-A", proxyPort, ingress, name)
 }
 
+func GetProxyPort(name string) uint16 {
+	prog := "iptables"
+	if !option.Config.EnableIPv4 {
+		prog = "ip6tables"
+	}
+
+	res, err := runProgCombinedOutput(prog, []string{"-t", "mangle", "-n", "-L", ciliumPreMangleChain}, false)
+	if err != nil {
+		return 0
```

This is supposed to be triggered after iptables initialization, right? So it should always succeed? By implication, perhaps we should add a warning here just so that if any of those constraints are not met, then we'll at least have a log telling us that we're silently failing to restore the proxy port due to the lack of chain.

---

I guess the flip side is that we have an "info" log describing which port we select when we *do* succeed, so maybe this is not necessary.

---

This is only supposed to work if the **previous** run installed rules, so this should work before any iptables initialization on the current run.